clients/python/moondream/cloud_vl.py

```diff
@@ -0,0 +1,161 @@
+import json
+import os
+import urllib.request
+import base64
+from PIL import Image
+from io import BytesIO
+from typing import Union, Optional, Literal
+
+
+from .types import (
+    VLM,
+    CaptionOutput,
+    EncodedImage,
+    QueryOutput,
+    DetectOutput,
+    SamplingSettings,
+)
+
+
+class CloudVL(VLM):
+    @classmethod
+    def from_api_key(
+        cls, api_key: str, api_url: str = None, version: str = "v1"
+    ) -> "CloudVL":
+        """Initialize a CloudVL instance with an API key.
+
+        Args:
+            api_key (str): The API key for authentication
+            api_url (str, optional): Custom API URL. Defaults to None.
+            version (str, optional): API version. Defaults to "v1".
+
+        Returns:
+            CloudVL: An initialized CloudVL instance
+        """
+        return cls(api_key=api_key, api_url=api_url, version=version)
+
+    def __init__(self, api_key: str = None, api_url: str = None, version: str = "v1"):
+        self.api_key = api_key or os.getenv("MOONDREAM_API_KEY")
+        if not self.api_key:
+            raise ValueError(
+                "API key is required. Set MOONDREAM_API_KEY env var or pass api_key"
+            )
+
+        self.api_url = api_url or f"https://api.moondream.ai/{version}"
+        self.client = urllib.request.urlopen
+        self.headers = {"X-Moondream-Auth": self.api_key}
+
+    def _encode_image(self, image: Image.Image) -> str:
+        try:
+            if image.mode != "RGBA":
+                image = image.convert("RGBA")
+            buffered = BytesIO()
+            image.save(buffered, format="PNG")
+            img_str = base64.b64encode(buffered.getvalue()).decode()
+            return f"data:image/png;base64,{img_str}"
+        except Exception as e:
+            raise ValueError(f"Failed to convert image to PNG: {str(e)}")
```

Comment:
Okay, this actually makes sense to do in encode_image. We may have to make it an abstract class?

Also, we don't support RGBA in the model, it always gets converted back to RGB. Let's use JPEG instead of PNG since it compresses better. Would also make sense to resize large images to have a maximum dimension of 768 to save the user's bandwidth.

Also I believe this is the recommended way to rethrow exceptions?

```python
try:
    frobnicate()
except KeyError as exc:
    raise ValueError("Bad grape") from exc
```